.github/workflows/generate-schema.yaml

```diff
@@ -0,0 +1,34 @@
+name: Generate Supergraph Schema
+
+on:
+  workflow_call:
+
+jobs:
+  generate_schema:
+    # Deduplicate jobs from pull requests and branch pushes within the same repo.
+    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.repository
```

Comment:
This de-duplication assumes we run on `pull_request` and `push`, so it should be included in the calling workflows instead of here